src/Provider.php

```diff
@@ -78,8 +79,17 @@ public function register(): void {
 		$this->container->singleton( Logger::class, Config::get_logger() );
 		$this->container->singleton( Tables_Provider::class );
 		$this->container->singleton( Regulator::class );
+
+		$prefix = Config::get_hook_prefix();
+
+		add_action(
+			"shepherd_{$prefix}_tables_registered",
+			function () {
```

Comment:
This use of the Closure will make it impossible to remove it after it's been registered on the action.
Is this by design? Is there a case where removing the `Regulator` class registration would be useful?
Since you cannot rely on a container that is guranteed to have the `instance` and `callback` methods like the DI52 does, the way you could make this registration removeable would be to create a `public function register_regulator` method on this class.

Reply:
Good comment! I move towards a register_regulator action